packages/daemon/src/host.js

```diff
@@ -159,6 +159,11 @@ export const makeHostMaker = ({
         deferTask(identifiers =>
           petStore.write(workerName, identifiers.workerFormulaIdentifier),
         );
+      } else {
+        const { type } = parseId(workerFormulaIdentifier);
+        if (type !== undefined && type !== 'worker') {
+          throw new Error(`Not a worker ${q(workerName)}`);
+        }
```

Comment:
Since we no longer store the formula type in the identifier, this refactor causes the test `'cannot spawn worker with existing non-worker name'` to fail. We can either mark that test as intentionally failing or attempt to address the matter here.

Reply:
Your discretion. If the test doesn’t work, delete it. I’m not entirely convinced it’s a feature.

Reply:
I think this needs to be resolved separately. I created #2147 and added a regression test for it in this PR.

Reply:
Resolution for this PR: We should simply fail if `foo` is not a worker. Whether `provideWorker()` overwrites the existing name is a separate question. We can create a regression test for the latter for now.